src/main/java/Joey.java

```diff
@@ -0,0 +1,137 @@
+            } else if (userCommand.startsWith("todo")) {
+                try {
+                    String description = userCommand.substring(5).trim();
+                    if (description.isEmpty()) {
+                        throw new JoeyException("Hey girlie, the description of the todo can't be empty. Follow this format instead: todo buy eggs");
+                    }
+                    tasks[taskCount] = new Todo(description);
+                    taskCount++;
+                    System.out.println(" Got it. I've added this task:");
+                    System.out.println("   " + tasks[taskCount - 1]);
+                    System.out.println(" Now you have " + taskCount + " tasks in the list.");
+                    System.out.println("____________________________________________________________");
+                } catch (StringIndexOutOfBoundsException e) {
+                    System.out.println("OOPS!!! The description of a todo cannot be empty.");
+                    System.out.println("____________________________________________________________");
+                } catch (JoeyException e) {
+                    System.out.println("OOPS!!! " + e.getMessage());
+                    System.out.println("____________________________________________________________");
+                }
+            } else if (userCommand.startsWith("deadline")) {
+                try {
+                    String[] parts = userCommand.substring(9).split("/by");
+                    if (parts.length != 2) {
+                        throw new JoeyException("Hey girlie, please provide both description and deadline for the deadline task. Follow this format instead: deadline return book /by Sunday");
+                    }
+
+                    String description = parts[0].trim();
+                    String by = parts[1].trim();
+
+                    if (description.isEmpty() || by.isEmpty()) {
+                        throw new JoeyException("Hey girlie, the description and deadline for the deadline task can't be empty. Follow this format instead: deadline return book /by Sunday");
+                    }
+
+                    tasks[taskCount] = new Deadline(description, by);
+                    taskCount++;
+                    System.out.println(" Got it. I've added this task:");
+                    System.out.println("   " + tasks[taskCount - 1]);
+                    System.out.println(" Now you have " + taskCount + " tasks in the list.");
+                    System.out.println("____________________________________________________________");
+                } catch (StringIndexOutOfBoundsException e) {
+                    System.out.println("OOPS!!! Hey girlie, please provide both description and deadline for the deadline task. Follow this format instead: deadline return book /by Sunday");
+                    System.out.println("____________________________________________________________");
```

Comment:
You might want to save this horizontal line in a variable and reference that instead of using this everywhere.